docs/tutorial.md

```diff
@@ -32,13 +32,122 @@ require('vendor/autoload.php');
 TODO: create client instance with discovery and do some requests
 ```
 
-## Handling errors
+## Using an asynchronous client
 
-TODO: explain how to handle exceptions, distinction between network exception and HttpException.
+When using an asynchronous client, it will use a PSR-7 `RequestInterface` and returns a `Http\Client\Promise` :
 
-## Doing parallel requests
+```php
+use Http\Discovery\HttpAsyncClientDiscovery;
 
-TODO explain sendRequests and how to work with BatchResult and BatchException
+$httpAsyncClient = HttpAsyncClientDiscovery::find();
+$promise = $httpAsyncClient->sendAsyncRequest($request);
+```
+
+This promise allows you to : 
+
+ * Add callbacks for when the response is available or an errors happens by using the then method:
```

Comment:
the part that comes after this bullet is very long. lets maybe do an overview list and then subsections with the code examples.